packages/commonwealth/client/scripts/views/pages/overview/topic_summary_row.tsx

```diff
@@ -135,28 +149,52 @@ export class ThreadsOverviewTopicSummaryRow
                         <CWText type="caption">+4 others</CWText>
                       </div> */}
                     </div>
-                    {/* <div class="row-bottom-menu">
-                      <CWIconButton
-                        iconSize="small"
-                        iconName="share"
-                        onclick={(e) => e.stopPropagation()}
-                      />
-                      <CWIconButton
-                        iconSize="small"
-                        iconName="flag"
-                        onclick={(e) => e.stopPropagation()}
-                      />
-                      <CWIconButton
-                        iconSize="small"
-                        iconName="bell"
-                        onclick={(e) => e.stopPropagation()}
-                      />
-                      <CWIconButton
-                        iconSize="small"
-                        iconName="dotsVertical"
-                        onclick={(e) => e.stopPropagation()}
-                      />
-                    </div> */}
+                    <div class="row-bottom-menu">
+                      <SharePopover />
+                      {isAdminOrMod && (
+                        <CWPopoverMenu
+                          menuItems={[
+                            {
+                              label: 'Delete',
+                              iconLeft: 'trash',
+                              onclick: async (e) => {
+                                e.preventDefault();
+
+                                const confirmed =
+                                  await confirmationModalWithText(
+                                    'Delete this entire thread?'
+                                  )();
+
+                                if (!confirmed) return;
+
+                                app.threads.delete(thread).then(() => {
+                                  navigateToSubpage('/');
```

Comment:
we should stay on the same page, wherever it is. if on overview, should stay on overview. if on /discussions, stay there